src/components/form/tickets/index.tsx

```diff
@@ -0,0 +1,123 @@
+import CustomSelect from '../../common/select';
+import DatePicker from '../../common/datePicker';
+import FormInput from '../../common/formInput';
+import { memo } from 'react';
+import { useAppDispatch } from '../../../store/hooks';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import {
+  EmptyHelperText,
+  FlexContainer,
+  HelperText,
+  StyledCancelButton,
+  StyledHeading,
+  StyledLoginButton,
+} from './styled';
+import { ITicketFieldValues, TicketsFormHelper, statusOptions } from './helper';
+import { ITicketState, ITickets, IUpdateTickets } from '../../../types/tickets';
+import { createTicketAsync, updateTicketAsync } from '../../../store/tickets/thunk';
+
+interface IProps {
+  toggleModal: () => void;
+  initialValues: ITicketState | null;
+  isEdit?: boolean;
+  apiUrl: string;
+}
+const TicketsForm = ({ toggleModal, initialValues, isEdit, apiUrl }: IProps) => {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    control,
+  } = useForm<ITicketFieldValues>({
+    defaultValues: {
+      ...initialValues,
+      deadlineDate: initialValues ? new Date(initialValues.deadlineDate) : '',
+    },
+  });
+
+  const dispatch = useAppDispatch();
+
+  const handleCreateTicket: SubmitHandler<ITickets> = async (data: ITickets) => {
```

Comment:
Let's make a common method with if else blocks

Reply:
fix: made a common method for Create & Edit